server/lib/db/transaction.ctrl.js

```diff
@@ -68,4 +119,24 @@ TransactionCtrl.gotTransaction = (err, transactions) => {
   }))
 }
 
+TransactionCtrl.internal.checkIfIdExists = (polymorphId) => {
+  return new Promise((fulfill, reject) => {
+    const query = TransactionModel.find()
+    try {
+      if (polymorphId) {
+        query.where('polymorph_id').equals(polymorphId)
+      }
+      query.exec()
+      .then((result) => {
+        if (result.length !== 0) {
+          fulfill(true)
```

Comment:
return also needed here, you will continue to call fulfill twice if length !== 0